qiskit_dynamics/signals/signals.py

```diff
@@ -406,12 +380,12 @@ def dt(self) -> float:
         return self._dt
 
     @property
-    def samples(self) -> Array:
+    def samples(self) -> ArrayLike:
         """
         Returns:
             samples: the samples of the piecewise constant signal.
         """
-        return Array(self._padded_samples[:-1])
+        return unp.asarray(self._padded_samples[:-1])
```

Comment:
This can be changed to 

```suggestion
        return self._padded_samples[:-1]
```
The reason we had `Array(self._padded_samples[:-1])` before was that `self._padded_samples` is an `Array`, but `self._padded_samples[:-1]` is actually the same as `self._padded_samples.data[:-1]` (i.e. it returns whatever the internal array is, rather than an `Array(self._padded_samples.data[:-1])`. Here we don't need to wrap or convert anything after indexing.

Reply:
Thanks. removed `unp.asarray` at https://github.com/Qiskit/qiskit-dynamics/pull/208/commits/d448b0ef14b6fff9b9890e787cb2c0698d590307.